straxen/plugins/events/event_s2_positions_flow.py

```diff
@@ -0,0 +1,314 @@
+        result["endtime"] = strax.endtime(events)
+
+        # Initialize contour fields with NaN values
+        for type_ in ["s2", "alt_s2"]:
+            result[f"{type_}_position_contour_flow_naive"] *= np.nan
+        result[f"position_contour_flow"] *= np.nan
+
+        # Split peaks by containment in events
+        split_peaks = strax.split_by_containment(peaks, events)
+
+        # Process each event
+        for event_i, (event, sp) in enumerate(zip(events, split_peaks)):
+            for type_ in ["s2", "alt_s2"]:
+                type_index = event[f"{type_}_index"]
+                if type_index != -1:
+                    # Store naive flow position contour
+                    result[f"{type_}_position_contour_flow_naive"][event_i] = sp[
+                        "position_contours_flow"
+                    ][type_index]
+
+                if type_ == "s2":
+                    if self.use_fdc_for_contour:
+                        # Apply full field distortion correction to contour
+                        contour_plus_xy = np.concatenate(
+                            [
+                                sp["position_contours_flow"][type_index],
+                                np.array([[sp["x_flow"][type_index], sp["y_flow"][type_index]]]),
+                            ],
+                            axis=0,
+                        )
+                        contour_with_z = np.concatenate(
+                            [
+                                contour_plus_xy,
+                                np.repeat(event["z_dv_corr"], self.n_poly + 2)[:, np.newaxis],
+                            ],
+                            axis=1,
+                        )
+                        delta_r = self.map(contour_with_z)
+                        scale = delta_r / np.linalg.norm(contour_with_z[:, :2], axis=1) + 1
+                        scaled_2d_contour = contour_with_z[:, :2] * scale[:, np.newaxis]
+
+                        # Store corrected contour and positions
+                        result["position_contour_flow"][event_i] = scaled_2d_contour[:-1]
+                        result["x_flow_fdc"][event_i] = scaled_2d_contour[-1, 0]
+                        result["y_flow_fdc"][event_i] = scaled_2d_contour[-1, 1]
+                    else:
+                        # Apply simple scaling based on field distortion correction
+                        scale = event["r_field_distortion_correction"] / event["r_naive"] + 1
+                        result["position_contour_flow"][event_i] = (
+                            sp["position_contours_flow"][type_index] * scale
+                        )
+                        result["x_flow_fdc"][event_i] = sp["x_flow"][type_index] * scale
+                        result["y_flow_fdc"][event_i] = sp["y_flow"][type_index] * scale
+
+        return result
+
+
+@export
+class EventPositionUncertainty(strax.Plugin):
+    """Plugin to calculate position uncertainties for events.
```

Comment:
Could you also add the relevant note in the docstring here? Same suggestion applied for other plugins too